### PR TITLE
Integration test for endpoints protected using Oauth(Password grant type)

### PR DESCRIPTION
**Pull request by thisaltennakoon**
Related PR https://github.com/wso2-support/product-apim/pull/1448

### CODE REVIEW

**Review by npamudika — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/AddEndPointSecurityPerTypeTestCase.java [hunk 1]
```diff
@@ -497,6 +496,113 @@ public void testAPIDefinitionImportWithEndpointSecurity() throws Exception {
                 applicationKeyBeanProduction.getConsumerSecret());
     }
 
+    @Test(groups = { "wso2.am"}, description = "Test add OAuth endpoint security for PASSWORD grant type",
+            dependsOnMethods = "testAddEndpointSecurityForOauthForClientCredentialsGrantType")
+    public void testAddEndpointSecurityForOauthForPasswordGrantType() throws Exception {
+
+        String clientCredGrantTypeEndpointSecurityForProductionAndSandboxForPasswordGrantType = "{\n" +
```
Comment (chameerar) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/AddEndPointSecurityPerTypeTestCase.java:503]: Better if we can make this string `final`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/ChangeEndPointSecurityPerTypeTestCase.java [hunk 2]
```diff
@@ -419,6 +419,98 @@ public void testUpdateEndpointSecurityForOauth() throws Exception {
         validateIntrospectionResponse(user, backendToken, applicationKeyBeanSandbox.getConsumerKey());
     }
 
+    @Test(groups = {
+            "wso2.am"}, description = "Test set endpoint security for PASSWORD grant type", dependsOnMethods =
+            "testUpdateEndpointSecurityForOauthForClientCredentialsGrantType")
+    public void testUpdateEndpointSecurityForOauthForPasswordGrantType() throws Exception {
+
+        String endpointSecurity = "{\n" +
```
Comment (chameerar) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/ChangeEndPointSecurityPerTypeTestCase.java:427]: Better if we can make this string `final`

**Comment by lakshithagunasekara** [comment]
Fixes: https://github.com/wso2/api-manager/issues/1100

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/AddEndPointSecurityPerTypeTestCase.java (hunk 1 above)
Comment (thisaltennakoon) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/AddEndPointSecurityPerTypeTestCase.java:503]: Fixed in https://github.com/wso2/product-apim/pull/13192

**Pull request merged**